emcc.py

```diff
@@ -1311,6 +1312,9 @@ def default_setting(name, new_default):
     for key in settings_map:
        if key not in COMPILE_TIME_SETTINGS:
          diagnostics.warning('unused-command-line-argument', "linker setting ignored during compilation: '%s'" % key)
+    for flag in LINK_TIME_FLAGS:
+        if flag in state.orig_args:
```

Comment:
This will loop over `orig_args` three times, which might be quite long. Instead, how about putting the link-only flags in a set, and looping over `orig_args` to just test each item for presence in that set?

Reply:
Good idea, done!